framework/auth/views.py

```diff
@@ -1123,8 +1126,8 @@ def external_login_email_post():
                 email=user.username
             )
             kind = 'success'
-            # 4. remove session and osf cookie
-            remove_session(session)
+            # 4. Clear session data
+            clear_external_first_login_anonymous_session_data(session)
```

Comment:
Fix 4:

Session cannot be removed since `push_status_message()` need to use that session. In stead clear "first-login" related information from the session. See `clear_external_first_login_anonymous_session_data()` for details.